### ISSUE TITLE
BaalHelper not looting

### ISSUE THREAD

**Issue by cameronpfs** [kolton/d2bot-with-kolbot#747]
After Diablo Helper finishes I added a delay before the pickit call so it would attempt looting, however, BaalHelper seems to leave as soon as the "assigned" leader leaves. Is there any way to get it to delay, so that it will take time to loot?

I have tried adding a delay before the pickit call but that doesn't seem to work.

**Comment by DarkHorseDre** [comment]
I've not done this but seems you want to delay the leader.  I'd edit the baal script and introduce a delay before the leader exits (do you mean exits the script, the game or the area?)

**Comment by cameronpfs** [comment]
I am using pubbotjoin with auto assign leader which works amazingly except for when the leader leaves without looting and I miss out on a lot of loot. 1st week back from a long vacation from d2, so haven't been able to thoroughly get gear and such to trade. The way I was looking at this is implementing something like if "leader" leaves then unassign leader; pickit; delay(500) move to quit or something like that, just not sure if it would work or not.

**Comment by cameronpfs** [comment]
I have already tried increasing that delay as well as placing a delay before the pickit call. What I mean is: 
bot currently does:
Load script;
assigns leader;
movetoposition;
precast;
takes portal;
kills throne room;
moves to Baal;
kills Baal;
leader exits;
bot exits.

what seems like is the leader exits so the bot exits too quickly for the pickit to run. anyway to catch that so it doesn't leave before picking up items?

Does that make more sense as to what I am trying to convey?

**Comment by DarkHorseDre** [comment]
Yes bro, so its back to what i said before: you want to delay the leader before they quit - I assume the leader is running the `baal.js` script, so put a delay in after he has killed baal and before he ends the script.

**Comment by cameronpfs** [comment]
So if I am not running the leader myself there is no way to do it then? I am using helper with auto leader in botpubjoin.

**Comment by DarkHorseDre** [comment]
ohhh shiz sorry bro - I misunderstood inbetween somewhere.

so its like my previois post: you will need to work out how to stop the bot checking for leader.. problem is that bot is designed to find leader and party and not play by itself.

i cant test this but could be around this part of logic (from baalhelper):
            if (party) {
                do {
                    if ((!Config.Leader || party.name === Config.Leader) && party.area === 131) {
                        break WSKLoop;

you would need to stop checking for party and leader once baal is dead, which happens here:
        Pather.moveTo(15134, 5923);
        Attack.kill(544); // Baal
        Pickit.pickItems();

you would need to go solo from before baal dies/as he dies as the leader usually leaves as soon as he has killed and picked items.

**Comment by DarkHorseDre** [comment]
I cant read the code well enough to know for sure what to change without testing - maybe someone else will chip in.

as such, what I would do is throw some console comments in to see exactly when the bot leaves.  I would change the party check to include a check for baal. something like (psuedo):

if not the leader or in party <usually exits> but am in throne and baal is dead then pickit
quit

of course youve got to find the correct part of the code which makes the bot exit, and I'm not sure if I was correct in my previous post..

**Comment by cameronpfs** [comment]
This is the config.killbaal section. If we could break party just before this point it would be fine. Would rather loot than xp.

`if (Config.BaalHelper.KillBaal) {
		Pather.moveTo(15092, 5011);
		Precast.doPrecast(false);

		while (getUnit(1, 543)) {
			delay(500);
		}

		delay(1000);
		Pather.moveTo(15092, 5011);

		portal = getUnit(2, 563);

		if (portal) {
			Pather.usePortal(null, null, portal);
		} else {
			throw new Error("Couldn't find portal.");
		}

		Pather.moveTo(15134, 5923);
		Attack.kill(544); // Baal
		Pickit.pickItems();
	}`

**Comment by DarkHorseDre** [comment]
Yeah the only thing I'm thinking is that if you comment out that section and just have it pickit once baal is dead, the leader leaves your bot will still quit if you haven't disabled the leader/party monitoring, which I am unsure of (may have a chance to look at this next week)

**Comment by cameronpfs** [comment]
By adding this after the pickit it wont do an immediate leave:

```
for (i = 0; i < Config.BaalHelper.Wait; i += 1) {
			party = getParty();

			if (party) {
				do {
					if ((!Config.Leader || party.name === Config.Leader) && party.area === 132) {
						break;
					}
				} while (party.getNext());
			}

			delay(1000);
		}
```

However, since it is a break it stops the script there and just stalls out if I could get it to exit properly at this point I would be okay.

**Comment by cameronpfs** [comment]
Is there a way to tell the console to pause; delay(1000); resume;  ?

**Comment by DarkHorseDre** [comment]
delay(millseconds) will pause the bot (I do it before opening TP for helpers as they can take a while to enter)

**Comment by DarkHorseDre** [comment]
i dont understand what you are asking: what is it you want to pause/delay?

**Comment by cameronpfs** [comment]
Nevermind, I was thinking if I could pause that script maybe I could do something like run just a pickit call but that would end up taking too long anyway. Just trying to figure out a way to stop it without it breaking. I was able to get the script to stop completely and it would pickit then stand there for however many minutes before stalling out. Do you know what I would need to put after it breaks to just leave or quit game?
